### PR TITLE
Refactor Progress Bar

### PR DESCRIPTION
**Pull request by ssandbekkhaug**
### Description

Motivation: Support showing a progress bar on the "Application Summary" page.

This PR:
- Extracts progress bar data and logic out of a Thymeleaf template into a Java class for safety
- Replaces existing uses of progressBar fragment with the new version
- Supports an "extra" step in the progress bar for the summary page

A following PR will add the progress bar to the summary page.

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [x] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)
- [x] Removed the release notes section if the title is sufficient for the release notes description, or put more details in that section.
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)
- [x] Extended the README / documentation, if necessary. For user-facing features, consider updating [the user docs](https://github.com/civiform/docs). For "under-the-hood" changes or things more relevant to developers, consider updating [the dev wiki](https://github.com/civiform/civiform/wiki).

#### User visible changes

- [x] Followed steps to [internationalize](https://github.com/civiform/civiform/wiki/Internationalization-%28i18n%29#application-strings) any new strings
  - [x] Added context strings to new [messages](https://github.com/civiform/civiform/blob/main/server/conf/i18n/messages)
  - [x] Didn't use a message in applicant facing code that isn't translated yet (unless behind a flag)
- [x] Wrote browser tests using the [validateAccessibility](https://sourcegraph.com/github.com/civiform/civiform/-/blob/browser-test/src/support/index.ts?L437:14&subtree=true) method
- [x] Made equivalent changes in Thymeleaf for applicant-facing features and changes (or created an issue in the [Northstar epic](https://github.com/orgs/civiform/projects/1/views/94) to track that it's missing from Thymeleaf)
- [x] Tested on mobile view. See [mobile device mode](https://developer.chrome.com/docs/devtools/device-mode/)
- [x] Manually tested at 200% size
- [x] Manually evaluated tab order
- [x] Manually tested with a screen reader if the feature is applicant-facing. See [screen reader testing](https://github.com/civiform/civiform/wiki/Testing#screen-reader-testing)

### Issue(s) this completes

Part of #8847

### CODE REVIEW

**Comment by ssandbekkhaug** [comment]
* **#8848** <a href="https://app.graphite.dev/github/pr/civiform/civiform/8848?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @ssandbekkhaug and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by rockycodes — APPROVED** [review]
nice refactor!

File: browser-test/image_snapshots/application_address_correction_test/north-star-verify-address-no-suggestions.png
[the remark this answers is not in the document]
Comment (gwendolyngoetz) [reply browser-test/image_snapshots/application_address_correction_test/north-star-verify-address-no-suggestions.png:1]: That looks pretty compressed. Not a blocking issue by any stretch, but please make sure Sijia has seen it. Thanks Don!

File: server/app/views/applicant/ApplicantBaseFragment.html [hunk 1]
```diff
@@ -39,21 +39,19 @@
   <span class="usa-hint--required">*</span>
 </th:block>
 
+<!-- The param is an instance of the ProgressBar class. -->
 <div
```
Comment (gwendolyngoetz) [inline server/app/views/applicant/ApplicantBaseFragment.html:43]: This should probably have an aria-role="[progressbar](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/progressbar_role)".

File: browser-test/image_snapshots/application_address_correction_test/north-star-verify-address-no-suggestions.png
[the remark this answers is not in the document]
Comment (ssandbekkhaug) [reply browser-test/image_snapshots/application_address_correction_test/north-star-verify-address-no-suggestions.png:1]: You're welcome! Just sent an email

File: server/app/views/applicant/ApplicantBaseFragment.html (hunk 1 above)
Comment (ssandbekkhaug) [reply server/app/views/applicant/ApplicantBaseFragment.html:43]: Will do. I added another "Done when" thing to https://github.com/civiform/civiform/issues/8847

**Pull request merged**